packages/flutter/lib/src/material/time.dart

```diff
@@ -48,20 +48,22 @@ enum DayPeriod {
 ///  * [DateTime], which represents date and time, and is subject to eras and
 ///    time zones.
 @immutable
-class TimeOfDay {
+class TimeOfDay implements Comparable<TimeOfDay> {
   /// Creates a time of day.
   ///
   /// The [hour] argument must be between 0 and 23, inclusive. The [minute]
   /// argument must be between 0 and 59, inclusive.
-  const TimeOfDay({ @required this.hour, @required this.minute });
+  const TimeOfDay({ @required int hour, @required int minute })
+      : assert(hour != null && hour >= 0 && minute != null && minute >= 0),
+        hour = hour % hoursPerDay,
```

Comment:
This is changing the behavior. Can we instead just assert that the values are in the right range as it is documented?